public/app/plugins/datasource/azuremonitor/components/TracesQueryEditor/setQueryValue.ts

```diff
@@ -0,0 +1,35 @@
+import { AzureMonitorQuery, ResultFormat } from '../../types';
+
+const buildTracesQuery = (operationId: string): string =>
+  `set truncationmaxrecords=10000;
+    set truncationmaxsize=67108864;
+    union isfuzzy=true traces, customEvents, pageViews, requests, dependencies, exceptions, availabilityResults
+    | where $__timeFilter()
+    | where (operation_Id != '' and operation_Id == '${operationId}') or (customDimensions.ai_legacyRootId != '' and customDimensions.ai_legacyRootId == '${operationId}')
+    | extend duration = iff(isnull(duration), toreal(0), duration)
+    | extend spanID = iff(itemType == "pageView" or isempty(id), tostring(new_guid()), id)
+    | extend serviceName = iff(isempty(name), column_ifexists("problemId", ""), name)
+    | project operation_Id, operation_ParentId, itemType, spanID, duration, timestamp, serviceName, customDimensions
+    | project-rename traceID = operation_Id, parentSpanID = operation_ParentId, startTime = timestamp, serviceTags = customDimensions`;
```

Comment:
Why is this being set here and not by the user? I think I'm not understanding how this all works

Reply:
This query is a generic query for retrieving a specific trace based on `operationId`. A user wouldn't need to really modify it to retrieve that trace. Also this is just the barebones editor at the moment so the finished product might look different 😉